File: LeetCode/1823. Find the Winner of the Circular Game/solution.cpp
/**
 * 95 / 95 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 5.8 MB 
 */
class Solution {
public:
    int findTheWinner(int n, int k) {
        int p = 0;
        for (int i = 2; i <= n; ++ i) {
            p = (p + k) % i;
        }
        return p + 1;
    }
};

File: LeetCode/1823. Find the Winner of the Circular Game/solution.py
"""
95 / 95 test cases passed.
Runtime: 260 ms
Memory Usage: 14.8 MB
"""
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        que = collections.deque(range(1, n + 1))
        for _ in range(n - 1):
            for i in range(k - 1):
                que.append(que.popleft())
            que.popleft()
        return que.popleft()
